server/tests/src/test/java/org/infinispan/server/functional/hotrod/HotRodFlagCacheOperations.java

```diff
@@ -198,6 +199,27 @@ public void testPublishers(EnumSet<Flag> flags) throws Throwable {
       cache.clear();
    }
 
+   @ParameterizedTest(name = TEST_OUTPUT)
+   @ArgumentsSource(FlagsProvider.class)
+   public void testKeyPublisher(EnumSet<Flag> flags) {
+      RemoteCache<String, String> cache = remoteCache(flags);
+
+      Map<String, String> content = new HashMap<>();
+      for (int i = 0; i < 25; i++) {
+         content.put("key-" + i, "value-" + i);
+      }
+
+      assertThat(cache.isEmpty()).isTrue();
+      cache.putAll(content);
+
+      CloseableIteratorSet<String> keys = cache.keySet();
+      for (String key : keys) {
+         assertThat(content.remove(key)).isNotNull();
```

Comment:
Hrmm.. shouldn't this be null if the FORCE_RETURN_VALUE flag wasn't set?

Reply:
`content` is a local `HashMap`. I remove from the local map and assert it is empty at the end.

Reply:
Ah, yes. Sorry, I was thinking of something else 👍